sale_commission/report/sale_commission_analysis_report.py

```diff
@@ -0,0 +1,102 @@
+from odoo import tools
+from odoo import models, fields, api
+from psycopg2.extensions import AsIs
+
+
+class SaleCommissionAnalysisReport(models.Model):
+    _name = "sale.commission.analysis.report"
+    _description = "Sale Commission Analysis Report"
+    _auto = False
+    _rec_name = 'commission_id'
+
+    invoice_state = fields.Selection([
```

Comment:
For not hardcoding this and not being tolerant to extensions, what you can do it put a `selection` method and use the same technique as here:

https://github.com/OCA/sale-workflow/blob/96956205fcd1474443f93da21aaff300f7bb0282/sale_order_type/models/sale_order_type.py#L16-L18